boards/nrf52840dk/Kconfig

```
@@ -16,4 +16,7 @@ config BOARD_NRF52840DK
     select HAS_PERIPH_USBDEV
     select HAS_VDD_LC_FILTER_REG0
 
+config MODULE_MTD_SPI_NOR
+    default y if MODULE_MTD
+
```

I think this is weakening the dependencies of the symbol. If we are doing it this way, we should add them here as well. Another option would be to have some 'feature' to indicate the default available backend.
```suggestion
config MODULE_MTD_SPI_NOR
    default y
    depends on MODULE_MTD
    depends on HAS_PERIPH_SPI
```

---

What do you mean by weakening?

---

When there are multiple definitions of a symbol, the dependencies are ORed, in this case it has no dependencies. From [here](https://docs.zephyrproject.org/latest/guides/build/kconfig/setting.html#configuring-invisible-kconfig-symbols):

>When defining a symbol in multiple locations, dependencies are ORed together rather than ANDed together. It is not possible to make the dependencies of a symbol more restrictive by defining it in multiple locations.

> When making changes to Kconfig.defconfig files, always check the symbol’s direct dependencies in one of the interactive configuration interfaces afterwards. It is often necessary to repeat dependencies from the base definition of the symbol to avoid weakening a symbol’s dependencies.